File: src/main/java/br/campos/poi/model/factory/DocumentFactory.java
package br.campos.poi.model.factory;

import br.campos.poi.model.abst.Document;
import br.campos.poi.model.abst.ObjectDTO;

/**
 * Fábrica de Documentos Mongo
 * 
 * @author Caio
 */
@FunctionalInterface
public interface DocumentFactory {

	/**
	 * Cria um Documento a partir de um DTO
	 * 
	 * @param dto DTO
	 * @return
	 */
	Document create(ObjectDTO dto);
}

File: src/main/kotlin/br/campos/poi/model/factory/DocumentFactory.kt
package br.campos.poi.model.factory

import br.campos.poi.model.abst.Document
import br.campos.poi.model.abst.ObjectDTO

/**
 * Fábrica de Documentos Mongo
 *
 * @author Caio
 */
fun interface DocumentFactory {

	/**
	 * Cria um Documento a partir de um DTO
	 */
	fun create(dto: ObjectDTO): Document
}
